be/src/http/action/tablet_migration_action.cpp

```diff
@@ -0,0 +1,258 @@
+                        {
+                            std::unique_lock<std::mutex> lock(_migration_status_mutex);
+                            std::map<MigrationTask, std::string>::iterator it_task =
+                                    _migration_tasks.find(current_task);
+                            if (it_task != _migration_tasks.end()) {
+                                it_task->second = "running";
+                            }
+                        }
+                        Status result_status = _execute_tablet_migration(tablet, dest_store);
+                        {
+                            std::unique_lock<std::mutex> lock(_migration_status_mutex);
+                            std::map<MigrationTask, std::string>::iterator it_task =
+                                    _migration_tasks.find(current_task);
+                            if (it_task != _migration_tasks.end()) {
+                                _migration_tasks.erase(it_task);
+                            }
+                            std::pair<MigrationTask, Status> finished_task =
+                                    make_pair(current_task, result_status);
+                            if (_finished_migration_tasks.size() >=
+                                config::finished_migration_tasks_size) {
+                                _finished_migration_tasks.pop_front();
+                            }
+                            _finished_migration_tasks.push_back(finished_task);
+                        }
+                    });
+                    if (!st.ok()) {
+                        status = Status::InternalError("Migration task submission failed");
+                        std::unique_lock<std::mutex> lock(_migration_status_mutex);
+                        std::map<MigrationTask, std::string>::iterator it_task =
+                                _migration_tasks.find(current_task);
+                        if (it_task != _migration_tasks.end()) {
+                            _migration_tasks.erase(it_task);
+                        }
+                    }
+                } while (0);
+            }
+            std::string status_result;
+            if (!status.ok()) {
+                status_result = to_json(status);
+            } else {
+                status_result =
+                        "{\"status\": \"Success\", \"msg\": \"migration task is successfully "
+                        "submitted.\"}";
+            }
+            req->add_output_header(HttpHeaders::CONTENT_TYPE, HEADER_JSON.c_str());
+            HttpChannel::send_reply(req, HttpStatus::OK, status_result);
+        } else {
+            DCHECK(goal == "status");
+            std::string status_result;
+            do {
+                std::unique_lock<std::mutex> lock(_migration_status_mutex);
+                std::map<MigrationTask, std::string>::iterator it_task = _migration_tasks.begin();
+                for (; it_task != _migration_tasks.end(); it_task++) {
```

Comment:
`_migration_tasks` is a map, so why not just find the task by key?
You can refer to `https://www.cnblogs.com/xupeidong/p/11976671.html`

Reply:
> `_migration_tasks` is a map, so why not just find the task by key?
> You can refer to `https://www.cnblogs.com/xupeidong/p/11976671.html`

OK. Thank you.